rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -221,6 +229,128 @@ def _decode_curve_deposit_events(
+            if (
+                tx_log.topics[0] == TOKEN_EXCHANGE and
+                pool_address in self.curve_pools and
+                len(self.curve_pools[pool_address]) > max(sold_token_id, bought_token_id)
+            ):
+                sold_token_address = self.curve_pools[pool_address][sold_token_id]
+                bought_token_address = self.curve_pools[pool_address][bought_token_id]
+        else:  # EXCHANGE_MULTIPLE
+            pool_or_router_address = CURVE_SWAP_ROUTER
+            spender_address = hex_or_bytes_to_address(tx_log.topics[1])
+            receiver_address = hex_or_bytes_to_address(tx_log.topics[2])
+            raw_sold_amount = hex_or_bytes_to_int(tx_log.data[-64:-32])
+            raw_bought_amount = hex_or_bytes_to_int(tx_log.data[-32:])
+            # Curve swap router logs route (a list of addresses) that was used. Route consists of
+            # at max 9 items. The structure is:
+            # 1. Token in
+            # 2..X Pools that were used
+            # X+1. Token out
+            # X+2..9 (unused items) Zero addresses
+            # Here we read only addresses of token in and token out.
+            sold_token_address = hex_or_bytes_to_address(tx_log.data[:32])
+            for i in range(1, 9):  # Starting from 1 because at 0 is `sold_token_address`
+                address = hex_or_bytes_to_address(tx_log.data[32 * i:32 * (i + 1)])
+                if address == ZERO_ADDRESS:
+                    break
+                bought_token_address = address
+
+        sold_asset: Optional[Asset] = None
+        if sold_token_address is not None:
+            if sold_token_address == ETH_SPECIAL_ADDRESS:
+                sold_asset = A_ETH
+            else:
+                sold_asset = Asset(evm_address_to_identifier(
+                    address=sold_token_address,
+                    chain_id=ChainID.ETHEREUM,
+                    token_type=EvmTokenKind.ERC20,
+                ))
+
+        bought_asset: Optional[Asset] = None
+        if bought_token_address is not None:
+            if bought_token_address == ETH_SPECIAL_ADDRESS:
+                bought_asset = A_ETH
+            else:
+                bought_asset = Asset(evm_address_to_identifier(
+                    address=bought_token_address,
+                    chain_id=ChainID.ETHEREUM,
+                    token_type=EvmTokenKind.ERC20,
+                ))
+
+        spend_event: Optional['EvmEvent'] = None
+        receive_event: Optional['EvmEvent'] = None
+        for event in decoded_events:
+            if event.address != pool_or_router_address:
+                continue
+
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+            if (
+                event.location_label == spender_address and
+                event.event_type == HistoryEventType.SPEND and
+                event.balance.amount == asset_normalized_value(amount=raw_sold_amount, asset=crypto_asset) and  # noqa: E501
```

Comment:
Can't do it since we don't know the asset for sure. In some cases (for example if a curve pool is not stored in out cache for some reason) there will be no asset known.